R/test-helpers.R

```diff
@@ -0,0 +1,68 @@
+# See https://r-pkgs.org/testing-advanced.html for explanation of having these
+# internal functions
+
+local_create_md <- function(temp_dir = tempdir(), env = parent.frame()) {
+  md_dir <- file.path(temp_dir, "markdown")
+  dir.create(md_dir, showWarnings = FALSE)
+  withr::defer(unlink(temp_dir), envir = env)
+
+  md_lines <- c(
+    "# H1",
+    "## H2",
+    "### H3",
+    "#### H4",
+    "__bold text__",
+    "",
+    "_italicized text_",
+    "",
+    "Numbered list",
+    "",
+    "1. First item",
+    "2. Second item",
+    "3. Third item",
+    "",
+    "Bulleted list",
+    "",
+    "- First item",
+    "- Second item",
+    "- Third item",
+    "",
+    "Here is some inline ````code````.",
+    "",
+    "[Markdown Guide](https://www.markdownguide.org)",
+    "",
+    "[````{nhsbsaShinyR}````](https://github.com/nhsbsa-data-analytics/nhsbsaShinyR)"
+  )
+
+  writeLines(
+    md_lines,
+    file.path(md_dir, "01_md_file.md")
+  )
+
+  writeLines(
+    md_lines,
+    file.path(md_dir, "02_md_file.md")
+  )
+
+  md_dir
+}
+
+
+local_create_word_doc <- function(temp_dir = tempdir(), md_dir, rv_dir, docx_file,
+                                  env = parent.frame()) {
+  withr::defer(unlink(temp_dir), envir = env)
+
+  styles_rmd <- system.file(
+    "review", "styles", "draft-styles.rmd",
+    package = "nhsbsaShinyR"
```

Comment:
I took your files in gpg report repo and realised that I have to change this to my package name (it kept failing test and found out this caused the issue) does it need to change?

Reply:
Get name dynamically. Note namespacing, and `methods` has been added to `DESCRIPTION`, even tho it is a built in package.

```
styles_rmd <- system.file(
    "review", "styles", "draft-styles.rmd",
    package = methods::getPackageName()
  )
```